src/devices.js

```diff
@@ -54,28 +38,21 @@ const getVRDisplay = async function (global) {
 };
 
 /**
- * Return polyfilled XRDevices based off of configuration
+ * Return polyfilled XR interface based off of configuration
  * and platform.
  *
  * @param {Object} global
  * @param {Object} config
- * @return {Promise<XRDevice?>}
+ * @return {Promise<PolyfilledXRDevice?>}
  */
-export const requestDevice = async function (global, config) {
-  // First, see if there are any native XRDevices on the platform
-  // in the case where we're not polyfilling the API, but providing
-  // a cardboard display if no native devices found.
-  let device = await getXRDevice(global);
-
-  if (device) {
-    return device;
-  }
-
-  // If no native XR devices found, check for a 1.1 VRDisplay.
+export const requestXRPolyfill = async function (global, config) {
+  let xr = null;
+  
+  // Check for a 1.1 VRDisplay.
   if (config.webvr) {
-    device = await getVRDisplay(global);
-    if (device) {
-      return new XRDevice(device);
+    xr = await getVRDisplay(global);
+    if (xr) {
+      return xr;
```

Comment:
Actually it's the reverse. Since the promise is being resolved within the XR object now, and we want that to be immediately available, the line futher down (wrapping the Cardboard device in `new XR()`) is wrong and this one was OK. The construction of the `XR` interface is happening in WebXRPolyfill.js